### PR TITLE
add functionality to step through macro expansion

### PR DESCRIPTION
**Pull request by KristofferC**
Fixes #152

### CODE REVIEW

File: src/commands.jl [hunk 1]
```diff
@@ -213,6 +213,35 @@ function next_line!(@nospecialize(recurse), frame::Frame, istoplevel::Bool=false
...
+    pc = maybe_step_through_macro_expansion!(frame)
+
+Return the current program counter of `frame` if it is not inside a macro expresion
+Otherwise, step through the statements of `frame` until the next statement is not inside
+a macro expansion.
+"""
+function maybe_step_through_macro_expansion!(@nospecialize(recurse), frame::Frame, istoplevel::Bool=false)
+    pc = frame.pc
+    is_macro_expansion(frame) || return pc
+    while pc !== nothing && !isa(pc, BreakpointRef)
```
Comment (KristofferC) [inline src/commands.jl]: This is kinda similar to `next_until!` but the predicate there takes an expression, but here we need more than the expression.

**Comment by KristofferC** [comment]
Does anyone know exactly when a 

For example:

```

julia> function f(x)
           x = sin(2)
           @test 1 == 1
           x = sin(2)
       end
f (generic function with 2 methods)

julia> (@code_lowered f(2)).linetable
3-element Array{Any,1}:
 Core.LineInfoNode(Main, :f, Symbol("REPL[16]"), 2, 0)
 Core.LineInfoNode(Main, :f, Symbol("REPL[16]"), 3, 0)
 Core.LineInfoNode(Main, :f, Symbol("REPL[16]"), 4, 0)
```

has no macro expansions line info.

**Comment by KristofferC** [comment]
It would be good to have a discussion of what should be the default here etc.

File: src/commands.jl [hunk 2]
```diff
@@ -213,6 +213,49 @@ function next_line!(@nospecialize(recurse), frame::Frame, istoplevel::Bool=false
 end
 next_line!(frame::Frame, istoplevel::Bool=false) = next_line!(finish_and_return!, frame, istoplevel)
 
+is_macro_expansion(lin::LineInfoNode) = lin.method == Symbol("macro expansion")
```
Comment (timholy) [inline src/commands.jl]: This function should be moved to the "predicates" section of `utils.jl`.

File: src/commands.jl [hunk 3]
```diff
@@ -213,6 +213,49 @@ function next_line!(@nospecialize(recurse), frame::Frame, istoplevel::Bool=false
...
+is_macro_expansion(::LineNumberNode) = false
```
Comment (timholy) [inline src/commands.jl]: Hmm, I might prefer this to be handled by `return isa(linfo, LineInfoNode) ? is_macro_expansion(linfo) : false` check in the method for `::Frame`. Wouldn't want this used on an expression that someone had manually `macroexpand`ed.

File: src/commands.jl [hunk 4]
```diff
@@ -213,6 +213,49 @@ function next_line!(@nospecialize(recurse), frame::Frame, istoplevel::Bool=false
...
+function is_macro_expansion(frame::Frame)
```
Comment (timholy) [inline src/commands.jl]: Possibly consider naming this `is_in_macro_expansion`?

File: src/commands.jl [hunk 5]
```diff
@@ -213,6 +213,49 @@ function next_line!(@nospecialize(recurse), frame::Frame, istoplevel::Bool=false
...
+    src = frame.framecode.src
+    codeloc = src.codelocs[frame.pc]
+    codeloc == 0 && return false
+    linfo = src.linetable[codeloc]
+    return is_macro_expansion(linfo)
+end
+
+"""
+    pc = maybe_step_through_macro_expansion!(recurse, frame)
```
Comment (timholy) [inline src/commands.jl]:
```suggestion
    pc = next_non_macro_expansion!(recurse, frame)
```

File: src/commands.jl [hunk 6]
```diff
@@ -213,6 +213,49 @@ function next_line!(@nospecialize(recurse), frame::Frame, istoplevel::Bool=false
...
+    pc = maybe_step_through_macro_expansion!(frame)
```
Comment (timholy) [inline src/commands.jl]:
```suggestion
    pc = next_non_macro_expansion!(frame)
```

**Comment by KristofferC** [comment]
Im not sure there is much worth salvaging here.

**Pull request closed without merging**